### PR TITLE
Fix deprecation messages for daal4py

### PR DESCRIPTION
**Pull request by Pahandrovich**

### CODE REVIEW

File: daal4py/sklearn/monkeypatch/dispatcher.py [hunk 1]
```diff
@@ -148,19 +148,26 @@ def enable(name=None, verbose=True, deprecation=True):
...
-    if verbose and sys.stderr is not None:
+        warnings.warn_explicit("\nScikit-learn patching with daal4py is deprecated "
+                               "and will be removed in the future.\n"
+                               "Use Intel(R) Extension "
+                               "for Scikit-learn* module instead "
+                               "(pip install scikit-learn-intelex).\n"
+                               "To enable patching, please use one of the "
+                               "following options:\n"
+                               "1) python -m sklearnex <your_script>\n"
+                               "2) Enable patching from your script:"
```
Comment (outoftardis) [inline daal4py/sklearn/monkeypatch/dispatcher.py:159]: maybe something like this is better (more consistent)

```suggestion
                               "1) From the command line:\n"
                               "    python -m sklearnex <your_script>\n"
                               "2) From your script:\n"
```

File: daal4py/sklearn/monkeypatch/dispatcher.py [hunk 2]
```diff
@@ -148,19 +148,27 @@ def enable(name=None, verbose=True, deprecation=True):
...
+                               "    patch_sklearn()",
+                               FutureWarning, "dispatcher.py", 151)
+    if verbose and deprecation and sys.stderr is not None:
         sys.stderr.write(
             "Intel(R) oneAPI Data Analytics Library solvers for sklearn enabled: "
             "https://intelpython.github.io/daal4py/sklearn.html\n")
+    if verbose and deprecation == False and sys.stderr is not None:
+        sys.stderr.write(
+            "Intel(R) Extension for Scikit-learn* solvers enabled: "
+            "https://intel.github.io/daal4py/scikit-learn-intelex\n")
```
Comment (PetrovKP) [inline daal4py/sklearn/monkeypatch/dispatcher.py:171]:
```suggestion
            "https://github.com/intel/scikit-learn-intelex\n")
```

**Comment by PetrovKP** [comment]
@Mergifyio backport rls/2021.2-rls

**Review by PetrovKP — APPROVED** [review]

File: daal4py/sklearn/monkeypatch/dispatcher.py [hunk 3]
```diff
@@ -148,19 +148,27 @@ def enable(name=None, verbose=True, deprecation=True):
...
+                               "    patch_sklearn()",
+                               FutureWarning, "dispatcher.py", 151)
+    if verbose and deprecation and sys.stderr is not None:
         sys.stderr.write(
             "Intel(R) oneAPI Data Analytics Library solvers for sklearn enabled: "
             "https://intelpython.github.io/daal4py/sklearn.html\n")
+    if verbose and not deprecation and sys.stderr is not None:
+        sys.stderr.write(
+            "Intel(R) Extension for Scikit-learn* solvers enabled: "
+            "https://github.com/intel/scikit-learn-intelex\n")
```
Comment (SmirnovEgorRu) [inline daal4py/sklearn/monkeypatch/dispatcher.py:171]:
```py
"Intel(R) Extension for Scikit-learn* enabled (https://github.com/intel/scikit-learn-intelex)\n"
```

File: setup_sklearnex.py [hunk 4]
```diff
@@ -37,8 +37,11 @@
 
 # sklearnex setup
 setup(name="scikit-learn-intelex",
-      description="Extension for scikit-learn with helps \
-            Intel(R) oneAPI Data Analytics Library",
+      description="Intel(R) Extension for Scikit-learn speeds up scikit-learn beyond by "
```
Comment (SmirnovEgorRu) [inline setup_sklearnex.py:40]: "Intel(R) Extension for Scikit-learn*" since it could be the first usage of full name in the code.

**Comment by PetrovKP** [comment]
@Mergifyio backport rls/2021.2.1-rls

**Pull request merged**